File: 2020_01_20/Nicitis/짝지어제거하기.cpp
#include <iostream>
#include <stack>

using namespace std;

int solution(string s)
{
	stack<char> charStack;

	for (int i = 0; i < s.length(); i++)
	{
		if (charStack.size() > 0 && s[i] == charStack.top())
			charStack.pop();
		else
			charStack.push(s[i]);
	}

	return charStack.size() == 0;
}

File: 2020_01_20/hookssi/짝지어 제거하기.py
def remove_couple(s):
    stack = []
    for i in range(0, len(s)):
        a = s[i]
        if(len(stack) > 0):
            b = stack.pop()
            if(a != b):
                stack.append(b)
                stack.append(a)
        else:
            stack.append(a)
            
    if(len(stack) > 0):
        return False
    else:
        return True
    
def solution(s):
    answer = list(s)
    return remove_couple(answer)
    
